### ISSUE TITLE
508 Defect 2, screen readers, incomplete form submission, error message vague and not read assertively

### ISSUE THREAD

**Issue by AngelaFowler82** [department-of-veterans-affairs/va.gov-team#25114]
Label: 24069

    - [508-defect-2]
     
    - https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2

Issue Title: 508 Defect 2, screen readers, incomplete form submission, error message vague and not read assertively.
Point of Contact: Angela Fowler
**VFS Point of Contact:** _First name only: Angela

<h2>User Story or Problem Statement:</h2>
As a screen reader user, I expect error messages to be read assertively and focus placed at the point of the error. As a user, I expect error messages to inform me as to the nature of the error and offer a solution.
<h2>Details</h2>
<p>When I change the board review option to "hearing" and do not specify a hearing type, the error message when I attempt to submit is not spoken assertively by my screen reader. I have to arrow around the screen to locate it. </p>
<p> The error message says that information is missing, but does not specify which information. This increases cognitive load and could cause confusion for users with cognitive disabilities. </p>
<h2>Acceptance Criteria</h2>
Error messages are informative and spoken assertively.
<h2>Environment</h2>
When the error is generated, NVDA announces "alert," while JAWS says nothing.
<h2>Steps to Recreate</h2>
<ul>
<li>Access 

https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal

and sign in with any test user. </li>
<li>Complete the form, initially stating you will upload evidence.</li>
<li>On the review screen, change the board review option to "hearing with a veterans law judge, do not specify the hearing type.</li>
<li>Attempt to submit</li>
</ul> 
![image](https://user-images.githubusercontent.com/78870095/119361933-095ef600-bc61-11eb-9272-d4dc35b7f64b.png)

**Comment by Mottie** [comment]
I believe the issue shown in the screenshot (no header highlighting) no longer exists. I completed the form selecting the upload choice, then switched the board review option to "hearing with a Veterans Law Judge" as directed.

![Highlighted board review option and hearing type headers](https://user-images.githubusercontent.com/136959/121212976-710d6780-c843-11eb-8831-dd9651a33451.png)

In the above screenshot, the "Board review option" and "Hearing type" headers are highlighted and include a warning icon, but I don't think these are set up to bring attention to a screenreader. Fixing this would require a change to the platform code globally for all forms.

The same generic alert shows up but doesn't specify the problem, but hopefully [the work in this PR will provide better error messaging](https://github.com/department-of-veterans-affairs/vets-website/pull/16431) once it is available, and updated for the Notice of Disagreement form.

**Comment by SarahKay8** [comment]
Hi @caw310 @Mottie May I get an updated link?

**Comment by SarahKay8** [comment]
@Mottie thnx!
